src/static/js/chapter.js

```diff
@@ -256,6 +256,102 @@ function setDiscussionCount() {
   }
 }
 
+function indexHighlighter() {
+    //Only activate this if IntersectionObserver is supported
+  if('IntersectionObserver' in window){
+    var indexScroller = document.querySelector('.index-scroller');
+
+    // Check if user has set reduced motion and only continue if not
+    var hasOSReducedMotion = window.matchMedia('(prefers-reduced-motion: reduce)').matches;
+    if (hasOSReducedMotion) {
+      console.log('User has set prefers-reduced-motion to ' + hasOSReducedMotion + ' so not highlighting the current section in chapter index');
+    } else {
+
+      // Check if 'position:sticky' is supported (as this is not great UX when not so don't bother)
+      // Add the sticky class (which sets 'position:sticky') and then test if that stuck :-)
+      indexScroller && indexScroller.parentNode && indexScroller.parentNode.classList.add('sticky');
+      var indexScrollerStyles = getComputedStyle(indexScroller.parentNode);
+      if (indexScrollerStyles && indexScrollerStyles.position) {
+        //Handle vendor prefixes (still required for Safari 12)
+        if (indexScrollerStyles.position.endsWith('sticky')) {
+
+          // Restrict the page height of the index to the page-height, as we're going to scroll this.
+          indexScroller.classList.add('page-height');
+
+          var indexItems = {};
+          var currentItem;
+
+          // Create a function to handle highlighting a new index item
+          // that will be called y bthe IntersectionObserver
+          function highlightIndexEntry(link) {
+            
+            var oldItem = currentItem;
+            var newItem = indexItems[link];
+            if (newItem){
+              newItem.classList.add('active');
+              if (oldItem && oldItem !== newItem) {
+                oldItem.classList.remove('active');
+              }
+              currentItem = newItem;
+
+              // If the index is too large to display in full then might need to change scroll
+              if (indexScroller.scrollHeight > indexScroller.clientHeight) {

```

Comment:
A simpler way to scroll to the active index item:

```js
index.scrollIntoView({behavior: 'smooth', block: 'center'})
```

We can do without the smooth scrolling if needed.

Reply:
Unfortunately can;'t use this as it scrolls the current element - which is fun when you have two scrolling elements like we do here! :-(

So works fine when clicking on the links in ToC, but not if focus is in main chapter text.